### PR TITLE
Release 2.0.0 (update all deps to latests stable)

### PR DESCRIPTION
**Pull request by JustBeard**
Update based on team decisions (except import/no-unused-modules - explanation provided in code).

Migration:
- add `"quoteProps": "as-needed"` to prettierrc
- new format for eslint globals
https://eslint.org/docs/user-guide/configuring#specifying-globals

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1 +1,11 @@
-module.exports = require('./index.js');
+const t = [require('./index.js')];
+
+module.exports = {
+  extends: require.resolve('./index.js'),
+  rules: {
+    // Forced to disable rule for eslint-config-eko as it behave weird.
+    // Somehow it reports all exports across package.
+    // Might be removed if we find out that it behave like this everywhere.
+    'import/no-unused-modules': 'off',
```
Comment (tavurth) [inline .eslintrc.js]: This rule should actually not report when the file has at least one export?

File: .eslintrc.js [hunk 2]
```diff
@@ -1 +1,11 @@
-module.exports = require('./index.js');
+const t = [require('./index.js')];
```
Comment (tavurth) [inline .eslintrc.js]: Why do you import this?

File: rules/import.js [hunk 3]
```diff
@@ -9,5 +10,50 @@ module.exports = {
...
+    //     unusedExports: true,
+    //   },
+    // ],
+
+    // Ref: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-useless-path-segments.md
+    //
+    // Airbnb
+    // 'import/no-useless-path-segments': ['error', { commonjs: true }],
+    //
+    // Re-defining airbnb to enable `noUselessIndex` option introduced with
```
Comment (tavurth) [inline rules/import.js:55]: prevents 

`test/test.js`
```js
import a from './../test/a.js' // (same directory)
```

File: README.md [hunk 4]
```diff
@@ -11,36 +11,24 @@ The package provides 2 sets of rules:
 
 Copy `.prettierrc` file into root of your project. Make sure that you do not change configuration or you will run in a bunch of conflicts between prettier and eslint.
 
-## To install base `eko` rules set:
+## Installation:
 
-### 1. Run npm insall for all required packages and `eslint-config-eko`:
+### 1. Insall `eslint-config-eko` and peer-dependencies:
```
Comment (cpv123) [inline README.md]: Misspelt (missing 't')

File: rules/import.js (hunk 3 above)
Comment (JustBeard) [reply rules/import.js:55]: @tavurth what do you mean?

Comment (JustBeard) [reply rules/import.js:55]: Is it argument against rule update or you suggestion to add this comment

File: .eslintrc.js (hunk 1 above)
Comment (JustBeard) [reply .eslintrc.js]: @tavurth 
```
- modules without any exports
- individual exports not being statically imported or requireed from other modules in the same project
```

Comment (JustBeard) [reply .eslintrc.js]: @tavurth 
I reverted `.eslintrc.js` as this code not required (I was forced to disable `import/no-unused-modules`)
https://github.com/EkoCommunications/EkoESLintConfig/pull/6/files#diff-17ec2f3ce30d31d97dd608a120ada07fR14

File: .eslintrc.js (hunk 2 above)
Comment (JustBeard) [reply .eslintrc.js]: `I reverted .eslintrc.js as this code not required (I was forced to disable import/no-unused-modules)
https://github.com/EkoCommunications/EkoESLintConfig/pull/6/files#diff-17ec2f3ce30d31d97dd608a120ada07fR14`

Originally I made this changes to disable import/no-unused-modules for eslint-config-eko.
We have `npm run test` which reads `.eslintrc.js ` and lint package.

File: rules/base.js [hunk 5]
```diff
@@ -53,23 +56,125 @@ module.exports = {
...
     // Airbnb: 'no-prototype-builtins': 'error',
-    // Example (if enable): obj.hasOwnProperty(key) => (should be) Object.prototype.hasOwnProperty.call(obj, key)
     //
-    // We prefer to get Error and get aware of the wrong usage. It is much better then silently hadle dangerous code.
+    // Example (if enable):
+    // obj.hasOwnProperty(key) => (should be) Object.prototype.hasOwnProperty.call(obj, key)
+    //
+    // We prefer to get Error and get aware of the wrong usage. It is much better then silently
+    // hadle dangerous code.
     'no-prototype-builtins': 'off',
```
Comment (frankent) [inline rules/base.js:70]: Should we enable it?

File: rules/react.js [hunk 6]
```diff
@@ -99,10 +104,73 @@ module.exports = {
...
+    'react/jsx-props-no-spreading': 'off',
+
+    // Ref: https://github.com/yannickcr/eslint-plugin-react/blob/bc976b837abeab1dffd90ac6168b746a83fc83cc/docs/rules/jsx-fragments.md
+    //
+    // Airbnb (since eslint-config-airbnb 18.0.0)
+    // 'react/jsx-fragments': ['error', 'syntax']
+    //
+    // Not an unanimous decision: enable but with `element` mode`
+    // Reasons: already existing preferences, more consistent.
+    'react/jsx-fragments': ['error', 'element'],
```
Comment (beermindnor) [inline rules/react.js:149]: <😭>I will get used to it </😭>

File: README.md [hunk 7]
```diff
@@ -11,35 +11,24 @@ The package provides 2 sets of rules:
...
 Copy `.prettierrc` file into root of your project. Make sure that you do not change configuration or you will run in a bunch of conflicts between prettier and eslint.
 
-## To install base `eko` rules set:
+## Installation:
 
-### 1. Run npm insall for all required packages and `eslint-config-eko`:
+### 1. Install `eslint-config-eko` and peer-dependencies:
 
-`npm i --save-dev eslint prettier eslint-config-airbnb-base eslint-config-prettier eslint-plugin-import eslint-plugin-prettier eslint-config-eko`
+(mandatory: base) `npx install-peerdeps --dev -x "--save-prefix=~" eslint-config-eko@2.0.0`
```
Comment (cpv123) [inline README.md:18]: I've run this command in a new repository and nothing gets installed. Am I doing something wrong or is it broken?

File: rules/base.js (hunk 5 above)
Comment (JustBeard) [reply rules/base.js:70]: We decided long time that we want to keep it disabled.

File: README.md (hunk 7 above)
Comment (JustBeard) [reply README.md:18]: eslint-config-eko@2.0.0 is not release yet. 
you can try with `eslint-config-eko@develop` or `eslint-config-eko@2.0.0-rc2`

File: rules/import.js (hunk 3 above)
Comment (tavurth) [reply rules/import.js:55]: Suggestion to add a comment, i had to look up the definition

File: rules/base.js (hunk 5 above)
Comment (JustBeard) [reply rules/base.js:70]: settled

**Review by cpv123 — APPROVED** [review]

**Review by beermindnor — APPROVED** [review]

**Review by Jedmed — APPROVED** [review]
LGTM

**Review by frankent — APPROVED** [review]

**Pull request merged**